pkg/controller/controller_pod.go

```diff
@@ -119,6 +121,22 @@ func (tc *TFJobController) reconcilePods(
 		// TODO(CPH): Need to delete pods.
 	}
 
+	if tfjob.Status.TFReplicaStatuses == nil {
+		tfjob.Status.TFReplicaStatuses = make(map[tfv1alpha2.TFReplicaType]*tfv1alpha2.TFReplicaStatus)
+	}
+
+	if _, ok := tfjob.Status.TFReplicaStatuses[rtype]; !ok {
+		tfjob.Status.TFReplicaStatuses[rtype] = &tfv1alpha2.TFReplicaStatus{}
+	}
+
+	tfjob.Status.TFReplicaStatuses[rtype].Active = int32(len(activePods))
+	tfjob.Status.TFReplicaStatuses[rtype].Succeeded = succeeded
+	tfjob.Status.TFReplicaStatuses[rtype].Failed = failed
+
+	if err := tc.updateTFJobStatus(tfjob); err != nil {
```

Comment:
I think we could add a TODO because there is no need to update the status every sync.

Reply:
Yes, done :)